mlflow/utils/databricks_utils.py

```diff
@@ -489,11 +501,36 @@ def get_model_dependency_oauth_token(should_retry=True):
             ) from e
 
 
+class TrackingURIConfigProvider(DatabricksConfigProvider):
+    """
+    TrackingURIConfigProvider extracts `scope` and `key_prefix` from tracking URI
+    of format like `databricks://scope:key_prefix`,
+    then read host and token value from dbutils secrets by key
+    "{key_prefix}-host" and "{key_prefix}-token"
+    """
+
+    def __init__(self, tracking_uri):
+        self.tracking_uri = tracking_uri
+
+    def get_config(self):
+        scope, key_prefix = get_db_info_from_uri(self.tracking_uri)
+
+        if scope and key_prefix:
+            dbutils = _get_dbutils()
+            if dbutils:
+                # Prefix differentiates users and is provided as path information in the URI
+                host = dbutils.secrets.get(scope=scope, key=key_prefix + "-host")
+                token = dbutils.secrets.get(scope=scope, key=key_prefix + "-token")
+                return DatabricksConfig.from_token(host=host, token=token, insecure=False)
+
+        return None
+
+
 def get_databricks_host_creds(server_uri=None):
```

Comment:
What about adding @deprecated warning (or at least some comment) to indicate this API shouldn't be used by external code base?🙂

Reply:
Make sense to me.